### PR TITLE
feat: simplify array UI by removing the dropdown menu

### PR DESCRIPTION
**Pull request by janechu**
# Description

This change updates the handling of the array data type in the `fast-form-generator-react` package to use a simplified method of controlling it and removes the drop down menu.

This change will go into a feature branch which also addresses a UI change for the handling of the React children data type.

## Motivation & context

This simplifies the UI for adding and removing array items.

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
[ ] This change causes current functionality to break.

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://microsoft.github.io/fast-dna/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-form-generator-react/src/form/form-item.array.style.ts [hunk 1]
```diff
@@ -20,59 +18,81 @@ const styles: ComponentStyles<FormItemArrayClassNameContract, {}> = {
...
+            content: "''",
+            opacity: ".6",
+            pointerEvents: "none",
+            top: toPx(11),
+            width: toPx(16),
+            height: toPx(16),
+            background: lines,
+            left: "0",
+        },
+        "& a": {
```
Comment (chrisdholt) [inline packages/fast-form-generator-react/src/form/form-item.array.style.ts]: @janechu do we not have a class for this? Would be *ideal* if we didn't have to target elements wholesale.

File: packages/fast-form-generator-react/src/form/form-item.children.style.ts [hunk 2]
```diff
@@ -46,9 +45,36 @@ const styles: ComponentStyles<FormItemChildrenClassNameContract, {}> = {
...
+            pointerEvents: "none",
+            top: "calc(50% - 8px)",
+            width: toPx(16),
+            height: toPx(16),
+            background: lines,
+            left: "0",
+        },
+        "& a": {
+            width: "calc(100% - 36px)",
+            "& span, i": {
```
Comment (chrisdholt) [inline packages/fast-form-generator-react/src/form/form-item.children.style.ts]: @janechu `li`? I don't think that we are using `i` (nor should we as that's a deprecated element).

Comment (janechu) [reply packages/fast-form-generator-react/src/form/form-item.children.style.ts]: Deprecated, where are you reading that? There is of course `<i>` and `<em>` which is similar but the text used here is not for emphasis, it's italicized to set it apart from the other text in the `<li>`.

See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/i

File: packages/fast-form-generator-react/src/form/form-item.array.style.ts (hunk 1 above)
Comment (janechu) [reply packages/fast-form-generator-react/src/form/form-item.array.style.ts]: We are very precisely targeting the `<a>` inside of the `<li>` which has the class `formItemArray_existingItemListItem`, there should only be a single `<a>` inside.

I can add a class and use that instead, however I would argue the reasoning is that it is more maintainable, I don't think it's a good argument that the targeting is "wholesale" in this case.

Comment (chrisdholt) [reply packages/fast-form-generator-react/src/form/form-item.array.style.ts]: Cool - primary consideration was the reasoning. Thx!

File: packages/fast-form-generator-react/src/form/form-item.children.style.ts (hunk 2 above)
Comment (chrisdholt) [reply packages/fast-form-generator-react/src/form/form-item.children.style.ts]: I wasn’t seeing the ‘i’ which is why I was curious. If we are considering this as a semantic representation for “italicized” text, the spec has changed: https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-i-element

I’d suggest if the existing spec doesn’t align and there is no semantic meaning - just presentational italics - we should just style with a span.

Comment (janechu) [reply packages/fast-form-generator-react/src/form/form-item.children.style.ts]: For context, the `<span>` is used as the name of the child component, the `<i>` is used if there is content for it, they should not be treated the same or in the "same voice" therefore the use of `<i>` to differentiate.

An example would be a component named "Button" with the inner text being "Hello world". You would then read "Button hello world" in the same voice if both used the `<span>` element.

Comment (chrisdholt) [reply packages/fast-form-generator-react/src/form/form-item.children.style.ts]: Fantastic, thanks for the context!

**Pull request merged**